cmd/metal-api/internal/service/machine-service.go

```diff
@@ -494,6 +505,52 @@ func (r machineResource) findMachine(request *restful.Request, response *restful
 	}
 }
 
+func (r machineResource) updateMachine(request *restful.Request, response *restful.Response) {
+	var requestPayload v1.MachineUpdateRequest
+	err := request.ReadEntity(&requestPayload)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	oldMachine, err := r.ds.FindMachineByID(requestPayload.ID)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	if oldMachine.Allocation == nil {
+		if checkError(request, response, utils.CurrentFuncName(), fmt.Errorf("only allocated machines can be updated")) {
+			return
+		}
+	}
+
+	newMachine := *oldMachine
+
+	if requestPayload.Name != nil {
+		newMachine.Name = *requestPayload.Name
```

Comment:
unsure if update name is a good idea

Reply:
Yes, I also think it can be dangerous in combination with Gardener. On the other hand when you use metalctl in combination with Gardener managed machines a user should know that this will always have dangerous implications.

Reply:
From pure perspective of the metal-api and metal machine provisioning I don't think it's dangerous, or?

Reply:
I would not enable users to do unnecessary and dangerous operations

Reply:
Removed. Editing tags is dangerous though, too! People could remove the Gardener cluster tag, which is important for the MCM.